ndcube/global_coords/global_coords.py

```diff
@@ -0,0 +1,72 @@
+from collections import OrderedDict
+from collections.abc import Mapping
+
+__all__ = ['GlobalCoords']
+
+
+class GlobalCoords(Mapping):
+    """
+    A structured representation of coordinate information applicable to a whole NDCube.
+    """
+    def __init__(self, ndcube):
+        """
+        Init method.
+        """
+        super().__init__()
+        self._ndcube = ndcube
+
+        self._internal_coords = OrderedDict()
+
+    @property
+    def _all_coords(self):
+        """
+        Establish _all_coords as a property returning some _internal_coords.
+        """
+        return self._internal_coords
+
+    def add(self, name, physical_type, coords):
+        """
+        Add a new coordinate to the collection.
+        """
+        self._internal_coords[name] = (physical_type, coords)
+
+    def remove(self, name):
+        """
+        Remove a coordinate from the collection
+        """
+        del self._internal_coords[name]
+
+    @property
+    def names(self):
+        """
+        A tuple of all the names or keys.
+        """
+        return tuple(self._all_coords.keys())
+
+    @property
+    def physical_types(self):
+        """
+        A tuple of all physical types, one per coordinate.
+        """
+        return tuple(item[0] for item in self._all_coords.values())
+
+    def __getitem__(self, item):
+        """
+        Index the collection by a name.
+        """
+        physical_type, value = self._all_coords[item]
+        gc = GlobalCoords(self._ndcube)
+        gc.add(item, physical_type, value)
+        return gc
+
+    def __iter__(self):
+        """
+        Iterate over the collection.
+        """
+        return iter(self._all_coords)
```

Comment:
I believe we want `iter` to just iterate through the coord values, not the physical types.  So I think this will need some more code.  I think the behaviour we want is:
```python
>>> gc = GlobalCoords()
>>> gc.add('name0', 'type0', 0)  # Where 0 takes the place of an actual coordinate value.
>>> gc.add('name1', 'type1', 1)
>>> for i in gc.items():  # Assuming GlobalCoords will have dictionary's items method. Not sure if it will
...     print(i)
('name0', 0)
('name1', 1)
```

Reply:
> Assuming GlobalCoords will have dictionary's items method. Not sure if it will

yes it should.

Reply:
Okay

Reply:
I did something similar

Reply:
This doesn't appear resolved.

Reply:
Revised and resolved

Reply:
Where is the change?

Reply:
Oh, it is in the test...

Reply:
```
for i, gc_item in enumerate(global_coords.keys()):
    if i == 0:
        assert (i, gc_item) == (0, 'name1')
    if i == 1:
        assert (i, gc_item) == (1, 'name2')
```

Reply:
This isn't what we need.

Reply:
Oh, okay... Think your expected behavior and the test are not entirely mutually exclusive.

Reply:
Let me double check for you...

Reply:
Should I remove the test? Think the existing code base got everything you needed for this PR covered.

Reply:
The way I proposed to test for `__iter__()` is only one way to do it. There can be others, or none at all.